scipy/optimize/_numdiff.py

```diff
@@ -206,6 +212,9 @@ def approx_derivative(fun, x0, method='3-point', rel_step=None, f0=None,
                      that the user function is real-valued and can be
                      analytically continued to the complex plane. Otherwise,
                      produces bogus results.
+
+        The method 'cs' is not available when `return_linear_operator`
```

Comment:
But as far as I understand we can still improve accuracy by using a complex step?

Basically `f(x + i*h*p) = f(x) + i h (J p)`, thus `J p = Im { f (x + i * h * p) } / h` --- such that we eliminated the difference and improved the round-off error.